core/src/main/java/com/linecorp/armeria/client/TlsProviderBuilder.java

```diff
@@ -0,0 +1,19 @@
+package com.linecorp.armeria.client;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class TlsProviderBuilder {
+    private final Map<String, TlsKeyPair> hostnameTlsKeyPair = new HashMap<>();
+
+    TlsProviderBuilder() {}
+
+    public TlsProviderBuilder addHostnameTlsKeyPair(String hostname, TlsKeyPair tlsKeyPair) {
```

Comment:
A wildcard certificate could be used to verify several hosts.
Should we change the implementation to satisfy the following specifications?
```java
TlsProvider
  // All *.a.com uses fooTlsKeyPair except foo.a.com
  .builder("*.a.com", fooTlsKeyPair)
  // Only foo.a.com uses barFooTlsKeyPair
  .builder("foo.a.com", barFooTlsKeyPair)
  // Only b.com uses bTlsKeyPair
  .builder("b.com", bTlsKeyPair)
  .build()
```